src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -1320,21 +1321,27 @@ synchronized public void onChangePredicate(Chat chat, String predicateName, Stri
     for (Session s : sessions.values()) {
       if (s.chat == chat) {
         // found session saving predicates
-        invoke("publishChangePredicate", s, chat, predicateName, result);
+        invoke("publishPredicate", s, chat, predicateName, result);
         s.savePredicates();
         return;
       }
     }
     error("could not find session to save predicates");
   }
 
-  public PredicateEvent publishChangePredicate(Session session, Chat chat, String name, String value) {
+  public PredicateEvent publishPredicate(Session session, Chat chat, String name, String value) {
     PredicateEvent event = new PredicateEvent();
     event.id = String.format("%s<->%s", session.userName, session.botInfo.name);
     event.userName = session.userName;
     event.botName = session.botInfo.name;
     event.name = name;
     event.value = value;
+    
+    if (name.equals("topic")) {

```

Comment:
Should probably do `"topic".equals(name)` to avoid NPE

Reply:
agree